### ISSUE TITLE
Strapi with docker-compose is unable to reinstall node_modules

### ISSUE THREAD

**Issue by MartinN3** [strapi/strapi-docker#225]
EDIT: MORE CONTEXT

Problem is that when you actually develop strapi, you commit app without node_modules and other trash. When you checkout that repo and do `docker-compose up` the strapi is unable to install missing NODE_MODULES ..... 

Filesystem volume snapshot:
```
docker-compose.yaml
app
    api
    components
    config
    exports
    extensions
    public
    .editorconfig
    .eslintignore
    .eslintrc
    .gitignore
    favicon.ico
    package.json
    README.md
```

```
version: "3"

services:
  strapi:
    image: strapi/strapi:3.0.5-node12-alpine
    container_name: strapi
    restart: unless-stopped
    environment:
      DATABASE_CLIENT: ${DATABASE_CLIENT}
      DATABASE_NAME: ${DATABASE_NAME}
      DATABASE_HOST: ${DATABASE_HOST}
      DATABASE_PORT: ${DATABASE_PORT}
      DATABASE_USERNAME: ${DATABASE_USERNAME}
      DATABASE_PASSWORD: ${DATABASE_PASSWORD}
    networks:
      - strapi-app-network
    volumes:
      - ./app:/srv/app
    ports:
      - "3000:1337"

etc.....
```

docker-compose up

```
strapi | Node modules not installed. Installing...
strapi | Starting your app...
strapi | Error loading the local develop command. Strapi might not be installed in your "node_modules". You may need to run "npm install"
```

**Comment by alexandrebodin** [comment]
Hello, Your issue does not give enough context to help you. Please add informations about your configuration and what your are doing. Closing until you change the description

**Comment by MartinN3** [comment]
@alexandrebodin is it now suitable for reopen?

**Comment by MartinN3** [comment]
I found issue and its related to bad managment on user side (not keeping yarn.lock or package.lock)

In this file /strapi/docker-entrypoint.sh

There is `if` and `elseif`, but `else` is missing.. therefore no response no installation nothing. only that message about running npm install..

`elif [ ! -d "node_modules" ] || [ ! "$(ls -qAL node_modules 2>/dev/null)" ]; then

    echo "Node modules not installed. Installing..."

    if [ -f "yarn.lock" ]; then

      yarn install

    elif [ -f "package-lock.json" ]; then

      npm install

    fi

  fi`

What suggestion about solving it do you guys have? Id use yarn by default because its wired deeply in strapi with its CLI. Or maybe fail with error that is more explaining?

**Comment by alexandrebodin** [comment]
I would do npm install if there is a package-lock and yarn install otherwise indeed

**Comment by alexandrebodin** [comment]
Just pushed a fix for this and update the 3.1.1 image. Closing this issue now
